readme.md

```diff
@@ -0,0 +1,305 @@
+- if used, the workflow file is required to be named `generate.mwe2` and placed in the top-level directory `workflow`
+- the workflow should place generated files in `${project.build.directory}/generated-sources`
+
+### `org.eclipse.xtext:xtext-maven-plugin`
+
+- the execution phase `generate-sources` is used to generate Java or Xtend code from code written in a Vitruvius DSL
+- if Xtend code is referenced from the DSL and vice versa, the Xtend code needs to be processed by the xtext-maven-plugin as well, in this case the `xtend-maven-plugin` should be disabled
+- projects using the plugin need to configure the DSL (see below)
+
+### `org.eclipse.xtend:xtend-maven-plugin`
+
+- used to generate Java code from (production and test) Xtend code
+- the generated code is placed in `${project.build.directory}/generated-sources/xtend` or `${project.build.directory}/generated-test-sources/xtend`
+
+### `maven-jar-plugin`
+
+- packages the build results in a jar archive
+- should be disabled for test-only projects by setting an empty `phase` for the execution `default-jar`
+- when packaging Ecore meta-models, the `MANIFEST.MF` file needs to be added to the jar, which is achieved by adding the following code to the plugin configuration
+
+```
+<archive>
+    <manifestFile>${project.basedir}/META-INF/MANIFEST.MF</manifestFile>
+</archive>
+```
+
+### `maven-gpg-plugin`
+
+- signs generated jar archives for deployment
+- automatically enabled by the Maven profile `snapshot`
+
+### `maven-install-plugin`
+
+- copies the packaged build results in the users local repository
+- must be disabled if `maven-jar-plugin` is disabled
+
+## Expected Project Structure
+
+The expected project structure is an extension of the standard Maven project structure.
+Not all of the shown directories and files are required for every project, use as applicable.
+The `reactions` directory is an example for a directory, in which code written in a DSL supported by Vitruvius should be placed.
+
+The file `.project` is only required when code is generated from Ecore meta-models or Xtext grammars.
+The file `plugin.xml` is only required when working with Ecore meta-models and will contain generated extension point definitions.
+There can be additional extension point definitions, e.g., when using custom factories for testing purposes.
+
+```
+src/
+- main/
+    - ecore/
+    - java/
+    - reactions/
+    - xtend/
+    - xtext/
+    - resources/
+- test/
+    - java/
+    - xtend/
+    - resources/
+workflow/
+    - generate.mwe2
+.project
+plugin.xml
+pom.xml
+```
+
+### Working with Ecore Meta-Models
+
+The Maven build parent defines a process and a collection of Maven plugins for working with `.ecore` and `.genmodel` files as seamlessly as possible.
+It can, however, be the case, that small changes to the files are necessary, mostly due to the way Eclipse handles URIs.
+
+For `.genmodel` files:
```

Comment:
It might be beneficial to also link examples here.

Reply:
Would it be sufficient to reference the [EMF-Template](https://github.com/vitruv-tools/EMF-Template) project here?

Reply:
In general a reference to the template project but probably also to specific files or lines in files to find the relevant properties faster.